File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        #prefix and suffix sum problem
        tot = sum(nums)
        way = 0
        prefix = 0
        for i in range(len(nums) - 1):
            prefix += nums[i]
            suffix = tot - prefix
            if prefix >= suffix:
                way += 1
        return way

File: 2358-number-of-ways-to-split-array/2358-number-of-ways-to-split-array.java
import java.util.stream.LongStream;
class Solution {
    public int waysToSplitArray(int[] nums) {
        long tot = 0;
        for(int val: nums){
            tot += val;
        }
        long prefix = 0;
        int cnt = 0;
        for(int i=0;i<nums.length-1;i++){
            prefix += nums[i];
            long suffix = tot - prefix;
            if (prefix >= suffix){
                cnt++;
            }
        }
        return cnt;
    }
}